components/tikv_util/src/config.rs

```diff
@@ -1353,15 +1434,51 @@ mod tests {
             tmp_dir.canonicalize().unwrap().join("test1.dump")
         );
 
-        let path2 = format!("{}", tmp_dir.to_path_buf().join("test2").display());
-        assert!(canonicalize_path(&path2).is_ok());
-        ensure_dir_exist(&path2).unwrap();
-        let res_path2 = canonicalize_path(&path2).unwrap();
-        assert_eq!(
-            Path::new(&res_path2),
-            Path::new(&path2).canonicalize().unwrap()
-        );
+        let cases = vec![".", "/../../", "./../"];
```

Comment:
What about the cases like `../wal_dir` when the `wal_dir` folder has not been created?

Reply:
I think the tests below can cover it. We don't want to test it in current directory because it's hard to cross validate the result with std.